assets/js/googlesitekit/data/create-error-store.test.js

```diff
@@ -0,0 +1,221 @@
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getErrorForSelector', () => {
+			it( 'requires a selector param', () => {
+				expect( () => {
+					select.getErrorForSelector();
+				} ).toThrow( 'selectorName is required.' );
+			} );
+
+			it( 'returns undefined when error does not exist', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				const selectedError = select.getErrorForSelector( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				expect( selectedError ).toBeUndefined();
+			} );
+
+			it( 'returns the required error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				const selectedError = select.getErrorForSelector( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+
+				expect( selectedError ).toMatchObject( ERROR_SAMPLE_2.error );
+			} );
+		} );
+
+		describe( 'getErrorForAction', () => {
+			it( 'requires a selector param', () => {
+				expect( () => {
+					select.getErrorForAction();
+				} ).toThrow( 'actionName is required.' );
+			} );
+
+			it( 'returns undefined when error does not exist', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				const selectedError = select.getErrorForAction( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				expect( selectedError ).toBeUndefined();
+			} );
+
+			it( 'returns the specified error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				const selectedError = select.getErrorForAction( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+
+				expect( selectedError ).toMatchObject( ERROR_SAMPLE_2.error );
+			} );
+		} );
+
+		describe( 'getError', () => {
+			it( 'requires a baseName param', () => {
+				select.getError();
+
+				expect( store.getState().error ).toBeUndefined();
+			} );
+
+			it( 'returns the correct error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+
+				const error = select.getError( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				expect( error ).toMatchObject( ERROR_SAMPLE_2.error );
+			} );
+		} );
+
+		describe( 'getErrors', () => {
+			it( 'returns an empty array if there are no errors', () => {
+				const errors = select.getErrors();
+				expect( errors ).toEqual( [] );
+			} );
+
+			it( 'checks if we have only 1 error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				const errors = select.getErrors();
+				expect( errors ).toHaveLength( 1 );
+			} );
```

Comment:
Generally speaking, selectors that "check" either return a boolean or throw, so the description is a little off here.

After selecting the errors, it's okay to check the length, but we can be a bit more thorough as we have the error object as well. We can actually test both with `expect( errors ).toEqual( [ error ] )` since there should only be one.